File: haskell/bob/Bob.hs
--John Youngblood
--2/9/2015
module Bob where
import Data.Char (isLower, isLetter, isSpace)

responseFor :: String -> String
responseFor s | all isSpace s = "Fine. Be that way!"
              | not (any isLower s) && any isLetter s = "Whoa, chill out!"
              | last s == '?' = "Sure."
              | otherwise = "Whatever."

File: javascript/bob/bob.js
//John Youngblood
//3-09-2015

var Bob = function () { };

Bob.prototype.hey = function (input) {

    var output = "Whatever.";

    if (input === input.toUpperCase() && input !== input.toLowerCase())
        output = "Whoa, chill out!";
    else if (input.charAt(input.length - 1) === '?')
        output = "Sure."
    else if (input.trim() === "")
        output = "Fine. Be that way!"
    return output;
};

module.exports = Bob;